rocAL/rocAL/source/meta_data/meta_node_resize_mirror_normalize.cpp

```diff
@@ -0,0 +1,100 @@
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "meta_node_resize_mirror_normalize.h"
+
+void ResizeMirrorNormalizeMetaNode::initialize()
+{
+    _src_height_val.resize(_batch_size);
+    _src_width_val.resize(_batch_size);
+    _dst_width_val.resize(_batch_size);
+    _dst_height_val.resize(_batch_size);
+    _mirror_val.resize(_batch_size);
+}
+void ResizeMirrorNormalizeMetaNode::update_parameters(MetaDataBatch *input_meta_data, const bool segmentation)
+{
+    initialize();
+    if (_batch_size != input_meta_data->size())
+    {
+        _batch_size = input_meta_data->size();
+    }
+    _mirror = _node->return_mirror();
+    _src_width = _node->get_src_width();
+    _src_height = _node->get_src_height();
+    _dst_width = _node->get_dst_width();
+    _dst_height = _node->get_dst_height();
+
+    vxCopyArrayRange((vx_array)_mirror, 0, _batch_size, sizeof(uint), _mirror_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_src_width, 0, _batch_size, sizeof(uint), _src_width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_src_height, 0, _batch_size, sizeof(uint), _src_height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_dst_width, 0, _batch_size, sizeof(uint), _dst_width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_dst_height, 0, _batch_size, sizeof(uint), _dst_height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+
+    for (int i = 0; i < _batch_size; i++)
+    {
+        _dst_to_src_width_ratio = _dst_width_val[i] / float(_src_width_val[i]);
+        _dst_to_src_height_ratio = _dst_height_val[i] / float(_src_height_val[i]);
+        auto bb_count = input_meta_data->get_bb_labels_batch()[i].size();
+        BoundingBoxCords coords_buf;
+        BoundingBoxLabels labels_buf;
+        coords_buf.resize(bb_count);
+        labels_buf.resize(bb_count);
+        memcpy(labels_buf.data(), input_meta_data->get_bb_labels_batch()[i].data(), sizeof(int) * bb_count);
+        memcpy((void *)coords_buf.data(), input_meta_data->get_bb_cords_batch()[i].data(), input_meta_data->get_bb_cords_batch()[i].size() * sizeof(BoundingBoxCord));
+        BoundingBoxCords bb_coords;
+        BoundingBoxLabels bb_labels;
+        if (segmentation)
+        {
+            // auto ptr = mask_data;
+            auto mask_data_ptr = input_meta_data->get_mask_cords_batch()[i].data();
+            int mask_size = input_meta_data->get_mask_cords_batch()[i].size();
+            for (int idx = 0; idx < mask_size; idx += 2)
+            {
+                if(_mirror_val[i] == 1)
+                {
+                    mask_data_ptr[idx] = _dst_width_val[i] - (mask_data_ptr[idx] * _dst_to_src_width_ratio) - 1;
```

Comment:
Over here are we directly modifying the mask data ptr is that why we have not assigned back to input meta data at the end?

Reply:
Yes